controllers/admin/AdminOrdersController.php

```diff
@@ -2088,21 +2088,45 @@ public function ajaxProcessAddProductOnOrder()
             )));
         }
 
-        $old_cart_rules = Context::getContext()->cart->getCartRules();
+        $product_informations = $_POST['add_product'];
+        $invoice_informations = $_POST['add_invoice'] ?? [];
+
+        $result = $this->addProductToOrder($order, $product_informations, $invoice_informations, Tools::getValue('add_product_warehouse'));
+        die(json_encode(array(
```

Comment:
Could you use `ajaxRender` instead please?

Reply:
:ok_hand: